drush.php

```diff
@@ -38,17 +38,15 @@ function drush_main() {
     register_shutdown_function('drush_coverage_shutdown');
   }
 
-  // Load the Drush configuration files.
+  // Load the global Drush configuration files, and global Drush commands.
   drush_preflight();
 
-  /* Set up bootstrap object, so that
-   * - 'early' files can bootstrap when needed.
-   * - bootstrap constants are available.
-   */
-  $bootstrap_class = drush_get_option('bootstrap_class', 'Drush\Boot\DrupalBoot');
-  $bootstrap = new $bootstrap_class;
-  drush_set_context('DRUSH_BOOTSTRAP_OBJECT', $bootstrap);
-  $bootstrap->preflight();
+  // Find the selected site based on --root, --uri or cwd, and
+  // return the bootstrap object that goes with it.
+  $bootstrap = _drush_preflight_root();
```

Comment:
This is the part of the PR that is most important to me.  This and the corresponding code re-org that goes with it makes the Drush bootstrap much cleaner, and I'd rather not go keep the existing code around for all of the Drush 7 stable branch's life.